docs/proposals/14-advanced-scheduling.md

```diff
@@ -0,0 +1,79 @@
+* If there is at least one un-tolerated taint with effect `NoExecute` then the shoot will be evicted from the seed (if it is already running on the seed), and will not be scheduled onto the seed (if it is not yet running on the seed).
+ 
+## Migrating Shoots Away from Unhealthy Seeds
+
+There are situations in which a seed can't function normally and should be either deleted or repaired "offline". This could be made easier by migrating shoots away from unhealthy seeds to healthy ones automatically or semi-automatically. This scenario is triggered by detecting an unhealthy seed. Once such a seed is detected, it should be drained of all its shoots, which means that they should be evicted from the seed and scheduled to other, healthy seeds.
+
+### Marking Seeds as "Unschedulable"
+
+The first step in migrating shoots away from an unhealthy seed is marking the seed as being "unschedulable" via a special taint, for example `seed.gardener.cloud/unschedulable`, with a `NoSchedule` effect. Initially, this could simply be done manually, or perhaps assisted by a special `gardenctl` command, for example `gardenctl taint <seed> `. In addition, the `gardenctl drain <seed>` command described below should add the taint to the specified seed prior to actually evicting all currently scheduled shoots.
+
+Marking a seed as "unschedulable" prevents any new shoots to be scheduled onto this seed, due to the `NoSchedule` effect of the taint. It is useful for evicting shoots away from unhealthy seeds, but not limited to this use case. Operators might want to mark some seeds as "unschedulable" also in cases where a seed temporarily cannot accept more shoots.
+
+Later, detecting unhealthy seeds and marking them as "unschedulable" could also be done automatically. This could be achieved by extending the `gardenlet` seed controller. In addition to setting certain seed conditions, the seed controller could also add the relevant taints with `NoSchedule` effects. Such taints could include for example:
+
+* `seed.gardener.cloud/not-ready`: The seed is `NotReady`. 
+* `seed.gardener.cloud/over-capacity`: The seed's capacity for shoots has been exceeded, see [Ensuring Seeds Capacity for Shoots Is Not Exceeded](13-automated-seed-management.md#ensuring-seeds-capacity-for-shoots-is-not-exceeded).
+
+**Note:** The above taints are only meant as examples. In particular, the `seed.gardener.cloud/not-ready` taint could be valuable if in some cases shoots could be asked to tolerate a "not-ready" taint, similarly to pods tolerating not-ready nodes. If this is not the case, then such a taint should probably not be introduced. 
+
+### Evicting Shoots from Drained Seeds
+
+The next step in migrating shoots away from an unhealthy seed is actually draining the seed, which involves evicting all shoots from this seed. This could be achieved in one of the following ways:
+
+* By manually marking all seed shoots as orphaned, by deleting the `seedName` field in the shoot specification.
```

Comment:
We could think about doing this via a subresource (e.g. `shoots/evict`), just like the eviction subresource for pods. And then disallowing this change via a normal `UPDATE shoots` call.
The benefits would be:
- the action can be protected by RBAC and therefore restricted to operators and controllers
- the subresource could accept additional paramters (not sure, if there is a use case for such things upfront, maybe "graceful eviction"?)
- further usecases like `ShootDisruptionBudgets` could be implemented more easily and in a cleaner way

This is a bit theoretical, but I guess, it can potentially save us some headaches and give us more freedom in the future.
WDYT?